### PR TITLE
Added Pre- and Post-Install Hook

### LINKED ISSUE #18: [FR] Run `flux check --pre` as a pre-install hook
To ensure the cluster is compatible with Flux, we could use the `ghcr.io/fluxcd/flux-cli:v0.23.0` image in a Kubernetes Job that runs `flux check --pre`. The job could be configured as a pre-install hook.

### PR DESCRIPTION
**Pull request by MarcNor**
Signed-off-by: Marc Normann <marc.normann@extern.dkbcodefactory.com>

#### What this PR does / why we need it:
This PR will add a Pre- and Post-Install Hook for Flux to run job _flux check --pre_

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #18 

#### Checklist

- [x] [DCO](https://github.com/fluxcd-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] helm-docs are updated
- [x] Helm chart is tested

### CODE REVIEW

**Comment by MarcNor** [comment]
@stefanprodan Opened a new branch and therefore new PR, since I worked on main before and got problems on rebasing. Same code as friday and all tests went fine.

File: charts/flux2/templates/post-install-job.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: post-install-job
+        image:  {{ .Values.checker.image }}:{{ .Values.checker.tag }}
+        command: ["/usr/local/bin/flux", "check", "--namespace", {{ .Release.Namespace }}]
```
Comment (stefanprodan) [inline charts/flux2/templates/post-install-job.yaml:28]: This does the same thing as `helm install/upgrade --wait`, I would remove this check for now.

File: charts/flux2/templates/pre-install-job.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Release.Name }}-pre"
```
Comment (stefanprodan) [inline charts/flux2/templates/pre-install-job.yaml:4]:
```suggestion
  name: "{{ .Release.Name }}-flux-check"
```

File: charts/flux2/templates/pre-install-job.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    metadata:
+      name: "{{ .Release.Name }}"
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: pre-install-job
```
Comment (stefanprodan) [inline charts/flux2/templates/pre-install-job.yaml:26]:
```suggestion
      - name: flux-cli
```

File: charts/flux2/values.yaml [hunk 4]
```diff
@@ -13,6 +13,9 @@ installCRDs: true
 # eg: http://notification-controller.[NAMESPACE].svc.[CLUSTERDOMAIN]
 eventsaddr: http://notification-controller/
 
+checker:
```
Comment (stefanprodan) [inline charts/flux2/values.yaml:16]:
```suggestion
cli:
```

**Review by stefanprodan — COMMENTED** [review]
Please add the `cli` values to the readme table.

File: charts/flux2/templates/post-install-job.yaml (hunk 1 above)
Comment (MarcNor) [reply charts/flux2/templates/post-install-job.yaml]: Thanks for the comment, they are resolved.

File: charts/flux2/templates/pre-install-job.yaml (hunk 2 above)
Comment (MarcNor) [reply charts/flux2/templates/pre-install-job.yaml]: Thanks for the comment, they are resolved.

File: charts/flux2/templates/pre-install-job.yaml (hunk 3 above)
Comment (MarcNor) [reply charts/flux2/templates/pre-install-job.yaml]: Thanks for the comment, they are resolved.

File: charts/flux2/values.yaml (hunk 4 above)
Comment (MarcNor) [reply charts/flux2/values.yaml]: Thanks for the comment, they are resolved.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @MarcNor 🏅

**Pull request merged**